Source/WebCore/platform/graphics/cocoa/CMUtilities.mm

```diff
@@ -636,6 +817,41 @@ void attachColorSpaceToPixelBuffer(const PlatformVideoColorSpace& colorSpace, CV
     });
 }
 
+#if ENABLE(ENCRYPTED_MEDIA) && HAVE(AVCONTENTKEYSESSION)
+Vector<Ref<SharedBuffer>> getKeyIDs(CMFormatDescriptionRef description)
+{
+    if (!description)
+        return { };
+    if (RetainPtr trackEncryptionData = static_cast<CFDataRef>(PAL::CMFormatDescriptionGetExtension(description, CFSTR("CommonEncryptionTrackEncryptionBox")))) {
+        // AVStreamDataParser will attach the 'tenc' box to each sample, not including the leading
+        // size and boxType data. Extract the 'tenc' box and use that box to derive the sample's
+        // keyID.
+        auto length = CFDataGetLength(trackEncryptionData.get());
+        auto ptr = (void*)(CFDataGetBytePtr(trackEncryptionData.get()));
+        Ref destructorFunction = createSharedTask<void(void*)>([data = WTFMove(trackEncryptionData)] (void*) { UNUSED_PARAM(data); });
+        Ref trackEncryptionDataBuffer = ArrayBuffer::create(JSC::ArrayBufferContents(ptr, length, std::nullopt, WTFMove(destructorFunction)));
+
+        ISOTrackEncryptionBox trackEncryptionBox;
+        auto trackEncryptionView = JSC::DataView::create(WTFMove(trackEncryptionDataBuffer), 0, length);
+        if (!trackEncryptionBox.parseWithoutTypeAndSize(trackEncryptionView))
```

Comment:
It seems we should have ISOTrackEncryptionBox take a std::span instead of a DataView.

Reply:
I just.moved that code